### PR TITLE
Workflows page refresh

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description

Live preview: https://internal.circleci.com/docs/workflows/

* Full page refresh to improve readability, section ordering
* Added `when` in workflows content to this workflows page
* Added config snippets to all filters sections of workflows page and configuration reference
* Added more regex examples due to customer feedback this was a pain point for workflow filters
* Added API steps for job approval

In a separate PR I will add flow diagrams to show workflow states, and workflow filtering decisions

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/workflows.adoc [hunk 1]
```diff
@@ -188,16 +147,36 @@ workflows:
...
 [#holding-a-workflow-for-a-manual-approval]
 == Hold a workflow for a manual approval
 
-Configure a workflow to wait for manual approval before continuing using an `approval` job. Anyone who has push access to the repository can click the *Approve* button in the CircleCI web app to continue the workflow.
+Use an `approval` job to configure a workflow to wait for manual approval before continuing. Anyone who has push access to the repository can approve the job to continue the workflow. Approve either by using the *Approve* button in the CircleCI web app, or via the API.
+
+Some things to keep in mind when using manual approval in a workflow:
+
+* `approval` is a special job type that is configured when listing jobs under the `workflow` key.
+* The approval job name must be a unique and not used by any other job in your configuration. For example, your configured jobs, such as `build` or `test1` in the example below, would not be set as `type: approval`.
```
Comment (briceicle) [inline jekyll/_cci2/workflows.adoc:162]: Remove the `a` in "The approval job name must be **a** unique and not used by..."

Comment (briceicle) [reply jekyll/_cci2/workflows.adoc:162]:
```
For example, your configured jobs, such as `build` or `test1` in the example below, would not be set as `type: approval`.
```

I don't understand how this relates to approval job names being unique. The jobs `build` or `test1` in the example could  have `type: approval` and the workflow would run fine.

File: jekyll/_cci2/workflows.adoc [hunk 2]
```diff
@@ -362,16 +362,73 @@ workflows:
...
+    }
+}
+----
+
+[#using-filters-in-your-workflows]
+== Using filters in your workflows
+
+The following sections provide examples for using filters in your workflows to manage job execution.
+
+You can filter workflows by branch, git tag, both, or neither. Workflow filters for branches and tags have the keys `only` and `ignore`:
```
Comment (briceicle) [inline jekyll/_cci2/workflows.adoc:411]: Technically you cannot filter workflows by both branch and tag, only one can be applied. So I would remove "both" in this sentence.

File: jekyll/_cci2/workflows.adoc [hunk 3]
```diff
@@ -423,15 +487,17 @@ workflows:
               only: /^v.*/
 ----
 
-In the example below, two jobs are defined within the `build-n-deploy` workflow:
+---
+
+In the example below, two jobs are configured within the `build-deploy` workflow:
 
 * The `build` job runs for all branches and all tags.
 * The `deploy` job runs for no branches and only for tags starting with 'v'.
```
Comment (briceicle) [inline jekyll/_cci2/workflows.adoc:495]: This is incorrect, and in contradiction with the description under the "Using filters in your workflows" section
```
If both branch and tag filtering is configured and a push to your code includes both branch and tag information, the branch filters take precedence.
```

In the example, both branch and tag filtering is configured so only the branch filtering would apply.

Comment (rosieyohannan) [reply jekyll/_cci2/workflows.adoc:495]: Oh yeah! Ok, it sounds like the example this is saying it shows is not actually possible to configure? I will simplify and remove the branch filter as you suggest. Thanks for spotting this!

File: jekyll/_cci2/workflows.adoc [hunk 4]
```diff
@@ -188,16 +147,37 @@ workflows:
...
 
 [#holding-a-workflow-for-a-manual-approval]
 == Hold a workflow for a manual approval
 
-Configure a workflow to wait for manual approval before continuing using an `approval` job. Anyone who has push access to the repository can click the *Approve* button in the CircleCI web app to continue the workflow.
+Use an `approval` job to configure a workflow to wait for manual approval before continuing. Anyone who has push access to the repository can approve the job to continue the workflow. Approve either by using the *Approve* button in the CircleCI web app, or via the API.
+
+Some things to keep in mind when using manual approval in a workflow:
+
+* `hold` is a special job type that is configured when listing jobs under the `workflow` key. You do not need to define a `hold` type job in the `jobs` section of your configuration. If you do configure steps for a job that is given the `hold` type in the workflows section, the steps for that job will not be run. A `hold` job is _only_ used to hold the workflow for approval, not to run any work.
```
Comment (briceicle) [inline jekyll/_cci2/workflows.adoc:161]: `hold` is not a job type.  `approval` is the job type that can be configured. So I think referring to `hod` as a job type is confusing.

I think `hold` is better described as the UI label/representation of an approval job.

**Review by briceicle — APPROVED** [review]

File: jekyll/_cci2/configuration-reference.adoc [hunk 5]
```diff
@@ -2396,6 +2414,8 @@ CAUTION: The scheduled workflows feature is set to be deprecated. Using *schedul
 
 A workflow may have a `schedule` indicating it runs at a certain time, for example a nightly build that runs every day at 12am UTC:
 
+Line 4:
+
```
Comment (kelvintaywl) [inline jekyll/_cci2/configuration-reference.adoc:2418]:
```suggestion
```

I think this was perhaps committed by mistake?
@rosieyohannan

Comment (rosieyohannan) [reply jekyll/_cci2/configuration-reference.adoc:2418]: SO what I really wanted to do was highlight line four in the config snippet but line highlighting isn't working yet unfortunately so I added it above

Screenshot 2024-06-25 at 10 34 05

Comment (rosieyohannan) [reply jekyll/_cci2/configuration-reference.adoc]: I have removed for now until we have a better way

**Review by alanjclark — APPROVED** [review]

**Pull request merged**